drivers/spi/spi_ll_stm32.c

```diff
@@ -851,7 +851,41 @@ static int transceive_dma(const struct device *dev,
 		spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 2);
 	}
 
+	size_t total_len = MAX(data->ctx.tx_len, data->ctx.rx_len);
```

Comment:
I think this won't calculate the actual total size.

`tx_len` and `rx_len` refer to the length of the current buffer, but there can be more than 1. To calculate the total length, it would be necessary to iterate over the buffers in `current_tx` and `current_rx`, from 0 to `tx_count` and `rx_count` respectively, considering that some of them could be `NULL`.